cars/app/cars/models.py

```diff
@@ -7,16 +8,36 @@ class Customer(models.Model):
 
     def buy_car(self, car_id):
         car = Car.objects.get(id=car_id)
-        car.customer = self;
-        car.save()
+        self.money -= car.price
+        try:
+            self.save()
+            car.customer = self;
+            car.save()
+        except Exception as error:
```

Comment:
Exception is redundant here. 
You use exceptions on where you don't control flow fully. 
For example, when calling database, API of other services and so on. 
If the code is 100% your you know what you transmit as a parameter and what you do with it. Exception will never happen. 

You sae in DB right here. But DB library that you use already has this Exception. You don't need to wrap it up.